docs/reference/setup/install/docker.asciidoc

```diff
@@ -154,7 +154,7 @@ services:
       memlock:
         soft: -1
         hard: -1
-    mem_limit: 1g
+      mem_limit: 1g
```

Comment:
`mem_limit` is not a ulimit in the [docker-compose v2 format](https://docs.docker.com/compose/compose-file/compose-file-v2/#ulimits) so it shouldn't be a key inside the `ulimit:` hash.

Instead [it needs to be specified](https://docs.docker.com/compose/compose-file/compose-file-v2/#mem_limit) at the same indentation level as image, ulimit etc. as  it currently is.

I just freshly copied/pasted the [docker-compose.yml](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html) from [the elastic.co website](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html) and it worked fine with docker-compose 1.13.0.

Testing should be done by inspecting `/sys/fs/cgroup/memory/memory.limit_in_bytes`, e.g. I defined 3g in my `docker-compose.yml` and see:

```
$ docker exec -u root -ti elasticsearch1 bash -c 'cat /sys/fs/cgroup/memory/memory.limit_in_bytes'
3221225472
```

Moving the indentation to the right, as in your PR, causes `mem_limit` not be honored and the container receives the cgroup default (which is unlimited).

What is your `docker-compose` version?